sprintQuovantis/sprintParam/views.py

```diff
@@ -258,11 +260,213 @@ def delete(self, request):
+            vote_by = {"vote_by": request.data.get("user_id")}
+            request_data = request.data.get("vote_table")
+            for votes_dic in request_data:
+                votes_dic.update(sprint)
+                votes_dic.update(vote_by)
+                votes_obj = Votes.objects.filter(
+                    vote_by=votes_dic.get("vote_by"),
+                    parameter_id=votes_dic.get("parameter_id"),
+                    sprint_id=votes_dic.get("sprint_id")
+                ).first()
+                if votes_dic.get("vote_by") == votes_dic.get("vote_for"):
+                    return Response(
+                        {
+                            "Message": "You Can not vote your self",
+                        },
+                        status=status.HTTP_400_BAD_REQUEST
+                    )
+                else:
+                    if votes_obj is None:
+                        serializer = VoteSerializer(
+                            data=votes_dic
+                        )
+                        serializer.is_valid(raise_exception=True)
+                        serializer.save()
+                    else:
+                        return Response(
+                            {"Message": "User already vote this parameter please choice another one"},
+                            status=status.HTTP_400_BAD_REQUEST
+                        )
+            return Response(
+                {
+                    "Message": "User Vote Successfully",
+                    "data": request_data
+                },
+                status=status.HTTP_201_CREATED
+            )
+        except ValidationError:
+            logging.error("Validation Failed")
+            return Response(
+                {
+                    "Message": "Enter Validate credentials"
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
         except Exception as e:
             logging.error(e)
             return Response(
                 {
-                    "message": "no such parameter found",
+                    "Message": "Enter all credentials"
                 },
-                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR
             )
+
+    @verify_token
+    def get(self, request, id):
+        """
+        this method is created for fetching the voting data
+        :param id:
+        :param vote_for:
+        :param request: format of the request
+        :return: Response
+        """
+        try:
+            sprint_total = Votes.objects.filter(sprint_id=id,vote_by=request.data.get("user_id"))
+            serializer = VoteSerializer(sprint_total, many=True)
+            if serializer.data:
+                votes_parameters = defaultdict(list)
+                for vote_dic in serializer.data:
+                    parameter_id = vote_dic["parameter_id"]
+                    vote_by = vote_dic["vote_by"]
```

Comment:
I believe it should be vote_for here, not vote_by. We need information about whom the currently logged in user has voted for in all parameters in a sprint.